libs/testnorge-oauth2/src/main/java/no/nav/registre/testnorge/libs/oauth2/service/ClientCredentialGenerateAccessTokenService.java

```diff
@@ -50,17 +52,23 @@ public ClientCredentialGenerateAccessTokenService(
         this.webClient = builder.build();
     }
 
-    public AccessToken generateToken(ClientCredential clientCredential) {
-        tokenResolver.verifyAuthentication();
+    public AccessToken generateToken(ClientCredential remoteClientCredential, AccessScopes accessScopes) {
+
         if (!tokenResolver.isClientCredentials()) {
-            throw new RuntimeException("Jwt er ikke av typen client credentials.");
+            throw new BadCredentialsException("Kan ikke gjennomfore opprette Client Credentials fra On Behalf Of.");
```

Comment:
Den som er brukt litt lenger ned er nok bedre å bruke 😄 
"Kan ikke gjennomfore On Behalf of fra Client Credentials."